### ISSUE TITLE
Enhancement: pass axios to the plugin rather than initializing it within the plugin

### ISSUE THREAD

**Issue by FPurchess** [zengoma/vuex-jwt#4]
The current axios initialization contains some opinionated parameters (CSRF headers). Initializing axios within the plugin also has the drawback that the user is not able to change the axios defaults to his needs. It would be great if the user can simply pass axios to the plugin and then the plugin installs it's interceptor. This would give the user more control over the instance.

**Comment by zengoma** [comment]
I agree. We had this plugged into a Django backend so we made some assumptions that are not universal.
